pkgs/development/libraries/libdevil/default.nix

```
@@ -35,6 +35,8 @@ stdenv.mkDerivation (finalAttrs: {
 
   configureFlags = [ "--enable-ILU" "--enable-ILUT" ];
 
+  CXXFLAGS = lib.optionalString stdenv.isDarwin "-Wno-register";
```

```suggestion
  CXXFLAGS = lib.optionalString stdenv.cc.isClang "-Wno-register";
```
for the initial error at least